### PR TITLE
LTP: Passing ext4 block device to enclave 

### PR DESCRIPTION
**Pull request by shaikshavali1**
This test case causing oom killer to be invoked and causing
kernel panic. This is because the test case invokes test framework
function "tst_acquire_device" which creates a 256MB of .img file
and binds with a loop device. This is not supported because the
default size of LKL memory is set to 32M. This is kept low due to
the limited size of EPC (Enclave page cache) and to avoid page
cache being swapped out.
Also, the test case invokes "tst_mkfs" framework function which
inturn invokes mkfs utility command with the help of system() syscall.
It is recommended to use the root file system, but, this test case
needs a read-only filesystem to test one of the sub-test cases.
we don't have a read-only file system mounted in sgx-lkl.

To support the test cases which need a block device. A formatted
ext4 filesystem image is created and passed
to enclave as an extended disk. The test case passes the details about this
new block disk image in the application configuration file. The test framework
will pick this image details in the environment variable and use it in the test.
Test case changes link: lsds/ltp#49, lsds/ltp#59.

This PR also revert the PR:#741
This #741, creates a filesystem image in the master image.
because of this, each individual test case needs to be modified.

### CODE REVIEW

File: tests/ltp/batch.mk [hunk 1]
```diff
@@ -22,7 +26,10 @@ ESCALATE_CMD=sudo
 $(ALPINE_TAR):
 	curl -L -o "$@" "https://nl.alpinelinux.org/alpine/v$(ALPINE_MAJOR)/releases/$(ALPINE_ARCH)/alpine-minirootfs-$(ALPINE_VERSION)-$(ALPINE_ARCH).tar.gz"
 
-$(ROOT_FS): $(ALPINE_TAR) ../buildenv.sh
+$(LTP_TEST_MNT_IMG):
+	dd if=/dev/zero of=$(LTP_TEST_MNT_IMG) count=$(LTP_TEST_MNT_IMG_SIZE) bs=1M
+
```
Comment (hukoyu) [inline tests/ltp/batch.mk:44]: Why this doesn't have line
mkfs -t ext4 $(LTP_TEST_MNT_IMG)

File: tests/ltp/run_ltp_test.sh [hunk 2]
```diff
@@ -28,7 +28,10 @@ case "$test_mode" in
 esac
 test_class="ltp"
 
-SGX_LKL_RUN_CMD=( "$SGXLKL_STARTER" $run_flag sgxlkl-miniroot-fs.img )
+SGXLKL_LTP_TSTAPP_CFG="--enclave-config=ltp_tstapp_enclave_config.json"
+SGXLKL_LTP_HOST_CFG="--host-config=ltp_host_config.json"
+
+SGX_LKL_RUN_CMD=( "$SGXLKL_STARTER" $SGXLKL_LTP_HOST_CFG $SGXLKL_LTP_TSTAPP_CFG $run_flag sgxlkl-miniroot-fs.img )
```
Comment (hukoyu) [inline tests/ltp/run_ltp_test.sh]: Please put all env variables in double quotes similar to "$SGXLKL_STARTER"

**Review by hukoyu — CHANGES_REQUESTED** [review]
@shaikshavali1 you have missing command in batch.mk. Please address my comments and if it passes tests after these changes, in general it looks good.

**Review by davidchisnall — APPROVED** [review]

**Comment by SeanTAllen** [comment]
@shaikshavali1 this will need to be rebased (somewhat carefully) against oe_port as a few changes have happened with these files. 

See the following PRs that are merged or likely will be before this one:

- https://github.com/lsds/sgx-lkl/pull/738
- https://github.com/lsds/sgx-lkl/pull/735
- https://github.com/lsds/sgx-lkl/pull/746

**Comment by hukoyu** [comment]
@shaikshavali1  as @SeanTAllen  said PR https://github.com/lsds/sgx-lkl/pull/746 is merged and you have to change your PR and make changes in batch.mk only according to new file/folder logic. Please rebase and reapply your changes.

File: tests/ltp/run_ltp_test.sh (hunk 2 above)
Comment (shaikshavali1) [reply tests/ltp/run_ltp_test.sh]: done

File: tests/ltp/batch.mk (hunk 1 above)
Comment (shaikshavali1) [reply tests/ltp/batch.mk:44]: done

**Comment by shaikshavali1** [comment]
> @shaikshavali1 as @SeanTAllen said PR #746 is merged and you have to change your PR and make changes in batch.mk only according to new file/folder logic. Please rebase and reapply your changes.

@hukoyu, I rebased changes.

**Comment by SeanTAllen** [comment]
It looks like something is wrong with the rebase. There's a lot of unrelated changes in this PR now.

**Comment by shaikshavali1** [comment]
> It looks like something is wrong with the rebase. There's a lot of unrelated changes in this PR now.

I will raise the new PR. It looks it is not possible to fix it now

**Comment by shaikshavali1** [comment]
@hukoyu , This pull request is broken now. Raised a new PR https://github.com/lsds/sgx-lkl/pull/764. Please verify and close this.

**Comment by hukoyu** [comment]
Dupe of https://github.com/lsds/sgx-lkl/pull/764

**Pull request closed without merging**